### ISSUE TITLE
Contract setup failures on CI

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#1015]
We keep running into trouble during contract setup when running tests on CI.
This might actually be a problem in production as well, but I was unable to reproduce it locally so far.

```
Jan 04 02:49:49.239  INFO happy_path::harness: Running version: 0.3.0
Jan 04 02:49:49.292  INFO daemon::maker_inc_connections: Listening on 127.0.0.1:49420
Jan 04 02:49:49.294 DEBUG daemon: Maker actor system ready
Jan 04 02:49:49.339 DEBUG daemon: Taker actor system ready
Jan 04 02:49:49.340 DEBUG daemon::connection: No connection to the maker
Jan 04 02:49:49.340 DEBUG daemon::connection: Connecting to maker address=127.0.0.1:49420
Jan 04 02:49:49.344  INFO daemon::maker_inc_connections: Upgrade new connection taker_address=127.0.0.1:49423
Jan 04 02:49:49.351 DEBUG daemon::noise: Noise protocol responder handshake is complete
Jan 04 02:49:49.354 DEBUG daemon::noise: Noise protocol initiator handshake is complete
Jan 04 02:49:49.355  INFO daemon::maker_inc_connections: Connection upgrade successful taker_id=62e40d623c1d070f845ae24bbb1ee5921f384215ec2aa3b63692fd8e7356f15b taker_address=127.0.0.1:49423
Jan 04 02:49:49.356  INFO daemon::connection: Established connection to maker address=127.0.0.1:49420
Jan 04 02:49:49.359  INFO daemon::taker_cfd: Taking current order: Order { id: OrderId(Hyphenated(2b9ad100-eb5b-488c-a5d7-36806dd48c92)), trading_pair: BtcUsd, position: Short, price: Price(50000), min_quantity: Usd(5), max_quantity: Usd(100), leverage: Leverage(2), liquidation_price: Price(33333.333333333333333333333333), creation_timestamp: Timestamp(1641264589), settlement_interval: Duration { seconds: 86400, nanoseconds: 0 }, origin: Theirs, oracle_event_id: BitMexPriceEventId { timestamp: OffsetDateTime { utc_datetime: PrimitiveDateTime { date: Date { year: 2022, ordinal: 5 }, time: Time { hour: 3, minute: 0, second: 0, nanosecond: 0 } }, offset: UtcOffset { hours: 0, minutes: 0, seconds: 0 } }, digits: 20 }, fee_rate: 1 }
Jan 04 02:49:49.361 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:49.363 DEBUG daemon::maker_cfd: Taker wants to take an order taker_id=62e40d623c1d070f845ae24bbb1ee5921f384215ec2aa3b63692fd8e7356f15b quantity=5 order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:49.366 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.392 DEBUG daemon::maker_cfd: Maker accepts order order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.393  INFO daemon::setup_maker: Maker accepts an order order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.394  INFO daemon::setup_taker: Order got accepted order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.404 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.681 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:49:50.903  INFO daemon::setup_contract: Exchanged setup parameters
Jan 04 02:49:56.772  INFO daemon::setup_contract: Created CFD transactions
Jan 04 02:49:57.196  INFO daemon::setup_contract: Exchanged setup parameters
Jan 04 02:50:06.914  INFO daemon::setup_contract: Created CFD transactions
Jan 04 02:50:07.761  INFO daemon::setup_contract: Exchanged CFD transactions
Jan 04 02:50:44.114  INFO daemon::setup_contract: Verified all signatures
Jan 04 02:50:44.430  INFO daemon::setup_contract: Exchanged CFD transactions
Jan 04 02:51:19.181  INFO daemon::setup_contract: Verified all signatures
Jan 04 02:51:19.183  INFO daemon::setup_contract: Exchanged signed lock transaction
Jan 04 02:51:42.088  INFO daemon::setup_contract: Exchanged signed lock transaction
Jan 04 02:52:02.281 DEBUG daemon::connection: No connection to the maker
Jan 04 02:52:02.282 DEBUG daemon::connection: Connecting to maker address=127.0.0.1:49420
Jan 04 02:52:02.287 ERROR daemon::maker_inc_connections: Failed to read incoming messages from taker taker_id=62e40d623c1d070f845ae24bbb1ee5921f384215ec2aa3b63692fd8e7356f15b
Jan 04 02:52:02.287  INFO daemon::maker_inc_connections: Dropping connection taker_id=62e40d623c1d070f845ae24bbb1ee5921f384215ec2aa3b63692fd8e7356f15b
Jan 04 02:52:02.287  INFO daemon::maker_inc_connections: Upgrade new connection taker_address=127.0.0.1:49452
Jan 04 02:52:02.294 DEBUG daemon::noise: Noise protocol responder handshake is complete
Jan 04 02:52:02.296 DEBUG daemon::noise: Noise protocol initiator handshake is complete
Jan 04 02:52:02.693  INFO daemon::maker_inc_connections: Connection upgrade successful taker_id=62e40d623c1d070f845ae24bbb1ee5921f384215ec2aa3b63692fd8e7356f15b taker_address=127.0.0.1:49452
Jan 04 02:52:02.693  INFO daemon::connection: Established connection to maker address=127.0.0.1:49420
Jan 04 02:52:02.696  INFO daemon::process_manager: Setup complete, publishing on chain now
Jan 04 02:52:02.696  INFO daemon::process_manager: Lock transaction published with txid 64cec9e8dcc708acc335f3cc6636989a71eede678dad062a9899042dfcbe0995
Jan 04 02:52:03.426 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
Jan 04 02:52:42.109 ERROR daemon::model::cfd: Contract setup failed: Expected Msg3 within 60 seconds: deadline has elapsed
Jan 04 02:52:42.111 DEBUG daemon::projection: Unable to calculate profit/loss without current price order_id=2b9ad100-eb5b-488c-a5d7-36806dd48c92
thread 'collaboratively_close_an_open_cfd' panicked at 'assertion failed: `(left == right)`
  left: `SetupFailed`,
 right: `PendingOpen`: cfd state mismatch between maker and taker', daemon/tests/happy_path.rs:304:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

see: https://github.com/itchysats/itchysats/runs/4697643307?check_suite_focus=true

I have seen several runs with similar failures. 

I am not entirely sure why this happens. 
From what I can see:

The maker fails to read a message from the taker and drops the connection. The taker reconnects successfully, but runs into a timeout with the contract setup then. It appears that one of the parties manages to publish lock while the other party transitions fails the setup.

**Comment by klochowicz** [comment]
I would bet that the same old problem we had came back again, ie. the heartbeat timeout triggers. We should be clear about it in the logs, right now if the timeout is triggered the connection is dropped with `NoReason`.
I reduced the frequency somewhat with recently increasing the timeout, but it still sometimes happen. BTW I don't think it would trigger in production, as it seems to be unique to our debug mode (contract setup takes ages...) and to the fact that the tests use a single thread for all tasks. Might have a look into it tomorrow again (started working on it before the weekend)
